axelarnetwork/axelar-core
# fix(vald): broadcast error handling

Because vald's broadcaster does not wait for a transaction to be committed, there are cases where a transaction fails to execute but no error is detected/thrown. One case where this leads to a bug is when gas estimation is inaccurate, which can happen if simulating the execution of some transaction takes a "cheaper" sequence of instructions than the actual execution. This scenario is very likely to happen during voting, since txs that come before, upon, and after the voting is completed can all take different execution paths.

To fix this issue, this PR refactors the broadcaster interface so that the Broadcast function receives the client context. All event handlers that trigger a voting will provide a client context with the broadcast mode set to block.

In addition to fixing the aforementioned problem, this PR also introduces a few additional logging output, and changes the TSS module to only save the group recovery info after the voting has been completed.

## Description

## Todos


## Steps to Test

Just spin up a local cluster and generate/rotate a key to ensure all works well.

## Expected Behaviour

The key should be successfully rotated

## Other Notes
